src/lib/Microsoft.Health.Fhir.Ingest/Service/MeasurementFhirImportService.cs

```diff
@@ -198,7 +198,10 @@ private static (IEnumerable<IMeasurementGroup> groups, Dictionary<IMeasurement,
             .GroupBy(m => $"{m.DeviceId}-{m.Type}-{m.PatientId}-{m.EncounterId}-{m.CorrelationId}")
             .Select(g =>
             {
-                IList<Measurement> measurements = g.ToList();
+                // In case of multiple measurements with the same OccurrenceTimeUtc within the batch, take the latest (based on IngestionTimeUtc) measurement
+                // to represent the recent/updated event data at that occurence timestamp.
+                IList<Measurement> measurements = g.GroupBy(x => x.OccurrenceTimeUtc).Select(y => y.OrderByDescending(d => d.IngestionTimeUtc).First()).ToList();

```

Comment:
Suggestion:  I think this can be replaced with ``.Select(y => y.Max(d => d.IngestionTimeUtc)``` and will be slightly more efficient.  In 99.99%+ of cases it won't make a difference because there will be 1 or 2 values max in the group, but if there is ever a larger group you avoid needing to do the sort when all you want is the max value in the collection.

Reply:
Yeah I had actually tried that first, but that returns the max value and not the entry corresponding to the max value. MaxBy() would actually do that, but that is only supported in >.net6

Reply:
Thanks for the clarification.  We will be looking to upgrade to .NET 7 soon.  At that time we can look to optimize.  Thanks!